server/models/auth/updateUser.js

```diff
@@ -55,9 +60,14 @@ function updateUser({ username, body }) {
       query += `email=?,verified=?,`
       arr.push(email)
       arr.push(0)
-      query += `otp=?,otp_valid_upto=NOW()+INTERVAL 1 DAY `
+      query += `otp=?,otp_valid_upto=NOW()+INTERVAL 1 DAY,`
       arr.push(otp)
     }
+    if (query[query.length - 1] === ',') {
+      query = query.substr(0, query.length - 1)
+    } else {
+      reject('No update requested')
+    }
```

Comment:
To remove the last `,` if present, so that the query becomes valid.
And if no `,` is present, it means the body didn't contain any of the required fields.